Addressing misalignment of apps on the Developer's Dashboard

CamiMonte opened the pull request:
This pull request should address the issue of the apps becoming misaligned when more than 2 apps have been added to the Developer dashboard. It also seeks to enhance the responsiveness of the navigation bar for the same.

CamiMonte committed:
Addressing misalignment of apps on the Developer's Dashboard

This commit should address the issue of the apps becoming misaligned when more than 2 apps have been added to the developer dashboard.

matjames007 reviewed (approved):
I have not verified this myself - but the stylistic changes should not affect core functionality.

matjames007 commented:
@CamiMonte please merge develop into this branch and attempt to resolve any conflicts that may arise.

CamiMonte committed:
This commit should update the navigation bar by re-adding the padding/space between the links.

Commit:
Merge branch 'develop' into developer-app-ui-fix

* develop: (34 commits)
  I've moved all the duplicate code from each of the /api endpoints to common-util.js and implemented the resulting function in each of the resource endpoints.
  This commit implements 5 new aggregate functions (avg, max, min, var, and stdev) - will provide further details in the comments on this pull request.
  Update db.js
  modified the 'main.css for redoc to accomodate angular serving the doc spec. I also reverted the docs main description to the hack that was previously there
  Changed the <alert> tag to a <div> and removed the "type" attribute. Added the "alert" and "text-center" classes to implement a consistent look with the rest of the bootstrap alerts.
  This implements a bootstrap alert that informs the user if the username or email address he has selected is already taken.
  I cave hanged the feecback messages given, when changing password, from being just html to using angularui and bootstrap alerts, so there is better consistency throughhout the app
  I cave hanged the feecback messages given, when changing password, from being just html to using angularui and bootstrap alerts, so there is better consistency throughhout the app
  I chave hanged the feecback messages given, when changing password, from being just html to using angularui and boostrap alerts, so there is better consistency throughhout the app
  When changing a user password, incorrect entry in the 'Old Password' field will no longer generate a javascript alertbox saying, 'undefined'. Instead, there is now a feedback message stating 'SUCCESS: Your password has been changed.' or 'ERROR: Incorrect password provided. Your password was not changed.', below the 'Password' title in the profile view for successfull and unsuccessful change password attempts, in providing the old password.
  This commit addresses the validation for the date parameters (where previously it would have worked if only a start or end date was specified but would have failed if neither was provided - a validation mistake that has now been corrected). Also adjusted the commenting to reflect the change of the 'count' function name to 'sum'.
  This commit ensures that if a non-numeric limit or offset parameter is provided - the API will fail gracefully and provide a helpful error message.
  restyling the docs to accomodate the previous changes in structure
  setting the 'scroll-y-offset' in docs.html
  commit before merge
  Creating a new Controller with the responsibility of initializing the Redoc elements.  This should give angular more control in the ordering of items it fires.  Testing needed.
  Minor style changes to account for missing navigation active indicator.
  In order for the password to be changed - as a security requirement the old password must be supplied (and used in validation).  A new endpoint is used to allow for password change for a specified user.  The endpoint doesn't check for authentication as it requires user_id and the user's old password.  Absence of this will throw an error.  The endpoint expects old_password and new_password within the request body.
  I opted for a more efficient way of doing the documentation.  Instead of an iframe I included it into the application.  The challenge of manipulating the DOM object within the HEAD tag was removed by utilizing an angular library that was capable of switching out the stylesheet based on the css specified within the $routeProvider.  It is important to note that the calls to the server are reduced and the UI interaction is preserved.
  The user may now change the password they have chosen when viewing their info on the 'profile.html'. The '.current_user.us_password' variable is now cleared when a user has logged in. This is to ensure the 'New Password' field is cleared when the form for changing the password is loaded.
  ...

# Conflicts:
#	public/stylesheets/style.css

nickwxo commented:
@CamiMonte I just merged develop into this branch and resolved the resulting merge conflict. I also tested your fix and it seems to be working well but you should check it too and make sure it's still doing what you intended and that I didn't break anything somehow.

And if it's still working as you intended - it can be merged into develop. 👍
